src/mlpack/methods/neighbor_search/knn_main.cpp

```diff
@@ -78,6 +82,9 @@ PARAM_FLAG("single_mode", "If true, single-tree search is used (as opposed to "
     "dual-tree search).", "S");
 PARAM_DOUBLE_IN("epsilon", "If specified, will do approximate nearest neighbor "
     "search with given relative error.", "e", 0);
+PARAM_STRING_IN("effective_error", "If specified, will compare the results "
+    "against the provided distances file, and will print the average relative "
+    "error.", "E", "");
```

Comment:
@rcurtin , yes I agree. I added 2 new command line options in https://github.com/mlpack/mlpack/pull/747/commits/92155154bc8721055702ae3d22c33542f2f07a61 and https://github.com/mlpack/mlpack/pull/747/commits/3e79d273545cb5d5694f6c8e7370a0dbbd95e5e6:
```
PARAM_STRING_IN("true_distances_file", "File of true distances to compute "
    "the effective error (average relative error) (it is printed when -v is "
    "specified).", "D", "");
PARAM_STRING_IN("true_neighbors_file", "File of true neighbors to compute the "
    "recall (it is printed when -v is specified).", "T", "");
```
So "-T" will print the *Recall*, similar to "-t" for *LSH*.